File: PythonProject/figures.py
from abc import ABC, abstractmethod
import math

class Figure(ABC):
    @abstractmethod
    def area(self):
        pass
    @abstractmethod
    def perimeter(self):
        pass
    @abstractmethod
    def scalingBy(self, coef):
        pass

class Circle(Figure):
    def __init__(self, r = 1.0):
        self.radius = r if r > 0 else 0.1

    def __str__(self):
        return f'Circle({self.radius})'

    def area(self):
        return math.pi * self.radius ** 2

    def perimeter(self):
        return 2.0 * math.pi * self.radius

    def scalingBy(self, coef):
        if (coef <= 0.0):
            coef = 0.1
        self.radius *= coef
        return self

class Rectangle(Figure):
    def __init__(self, a = 3.0, b = 4.0):
        self.length = a if a > 0 else 0.1
        self.width = b if b > 0 else 0.1

    def __str__(self):
        return f'Rectangle[{self.length}x{self.width}]'

    def area(self):
        return self.length * self.width

    def perimeter(self):
        return 2.0 * (self.length + self.width)

    def scalingBy(self, coef):
        if (coef <= 0.0):
            coef = 0.1
        self.length *= coef
        self.width *= coef
        return self

File: CSharpProject/Figures.cs
using System;

namespace CSharpProject
{
    abstract class Figure /*: ICloneable*/
    {
        public abstract double area();
        public abstract double perimeter();
        public abstract Figure scalingBy(double coef);
        public abstract Figure Clone();
    }
    class Circle : Figure
    {
        private double radius;
        public Circle()
        {
            radius = 1.0;
        }
        public Circle(double r)
        {
            radius = r > 0.0 ? r : 0.1;
        }
        public override double area()
        {
            return Math.PI * radius * radius;
        }
        public override double perimeter()
        {
            return 2.0 * Math.PI * radius;
        }
        public override Circle scalingBy(double coef)
        {
            if (coef <= 0.0) coef = 0.1;
            radius *= coef;
            return this;
        }
        public override Circle Clone()
        {
            return new Circle(this.radius);
        }
        public override string ToString()
        {
            return $"Circle({radius})";
        }
    }
    class Rectangle : Figure
    {
        private double length, width;
        public Rectangle()
        {
            length = 3.0;
            width = 4.0;
        }
        public Rectangle(double a, double b)
        {
            length = a > 0.0 ? a : 0.1;
            width = b > 0.0 ? b : 0.1;
        }
        public override double area()
        {
            return length * width;
        }
        public override double perimeter()
        {
            return 2.0 * (length + width);
        }
        public override Rectangle scalingBy(double coef)
        {
            if (coef <= 0.0) coef = 0.1;
            length *= coef;
            width *= coef;
            return this;
        }
        public override Rectangle Clone()
        {
            return new Rectangle(this.length, this.width);
        }
        public override string ToString()
        {
            return $"Rectangle[{length}x{width}]";
        }
    }
}
